### PR TITLE
[1.x] chore: bump to final Python 3.5 pin

### PR DESCRIPTION
**Pull request by henryiii**
Using the final 3.5 pinned images as mentioned by @mayeut.

### CODE REVIEW

**Comment by mayeut** [comment]
My comment in #657 was misleading (it was for not having auditwheel 4.0).
To still get python 3.5:
`2021-05-01-28d233a` c.f. https://github.com/pypa/manylinux/issues/995#issuecomment-832425679

**Comment by henryiii** [comment]
Ahh, yes, I got the two combined.

File: cibuildwheel/resources/pinned_docker_images.cfg [hunk 1]
```diff
@@ -1,27 +1,27 @@
 [x86_64]
 manylinux1 = quay.io/pypa/manylinux1_x86_64:2021-04-05-14df3a6
```
Comment (mayeut) [inline cibuildwheel/resources/pinned_docker_images.cfg:2]:
```suggestion
manylinux1 = quay.io/pypa/manylinux1_x86_64:2021-05-05-b64d921
```
Commit just before `auditwheel==4.0.0`
Should we ?

File: cibuildwheel/resources/pinned_docker_images.cfg [hunk 2]
```diff
@@ -1,27 +1,27 @@
...
 manylinux2010 = quay.io/pypa/manylinux2010_x86_64:2021-02-06-3d322a5
-manylinux2014 = quay.io/pypa/manylinux2014_x86_64:2021-04-05-b4fd19d
-manylinux_2_24 = quay.io/pypa/manylinux_2_24_x86_64:2021-04-05-b4fd19d
+manylinux2014 = quay.io/pypa/manylinux2014_x86_64:2021-05-01-28d233a
+manylinux_2_24 = quay.io/pypa/manylinux_2_24_x86_64:2021-05-01-28d233a
 
 [i686]
 manylinux1 = quay.io/pypa/manylinux1_i686:2021-04-05-14df3a6
```
Comment (mayeut) [inline cibuildwheel/resources/pinned_docker_images.cfg:8]:
```suggestion
manylinux1 = quay.io/pypa/manylinux1_i686:2021-05-05-b64d921
```
Commit just before `auditwheel==4.0.0`
Should we ?

**Comment by joerick** [comment]
Does this need a 1.x release @henryiii ?

**Pull request merged**